syntax = "proto3" ;

service UserManagement {
    rpc CreateUser (CreateUserRequest) returns (CreateUserResponse);
    rpc GetUserDetails (UserId) returns (UserDetails); 
    rpc UpdateUserPassword (UpdatePassword) returns (Response);
    rpc DeleteUser (UserId) returns (Response);
    rpc GetUserId(GetUserIdRequest) returns (GetUserIdResponse);

}

message UpdatePassword {
    UserId user_id = 1;
    string old_password = 2;
    string new_password = 3;
}

message UserId {
    int32 id = 1;
}

message UserDetails {
    string username = 1;
    string email = 2;
}

message CreateUserRequest {
    string username = 1;
    string email = 2;
    string password = 3;
    string confirm_password=4;
}

message Response {
    string status = 1;
    string message = 2;
}

message CreateUserResponse {
    string status = 1;
    string message = 2;
    string username = 3;
    string email = 4;
    int32 id = 5;
}

message GetUserIdRequest {
    string email = 1;
}

message GetUserIdResponse {
        string status = 1;
    }
